### ISSUE TITLE
9gloiCKaggPKNHAz

### ISSUE THREAD

**Issue by 0xym** [Kunzisoft/KeePassDX#653]
**Describe the bug**

**To Reproduce**

**Expected behavior**

** Keepass Database **
 - Created with: [e.g Windows KeePass 2.42]
 - Version: [e.g. 2]
 - Location: [e.g. Remote file retrieved with GDrive app]
 - Size: [e.g. 150Mo]
 - Contains attachment: [e.g. Yes]

**KeePassDX (please complete the following information):**
 - Version: [e.g. 2.5.0.0beta23]
 - Build: [e.g. Free]
 - Language: [e.g. French]

**Android (please complete the following information):**
 - Device: [e.g. GalaxyS8]
 - Version: [e.g. 8.1]

**Additional context**
 - Browser for Autofill: [e.g. Chrome version X]

**Comment by J-Jamet** [comment]
I cannot reproduce your issue with your procedure.
On my devices, step 3 removes the entry from the Magikeyboard.
Are you sure **_"Settings -> Magikeyboard settings -> Clear at closing"_** is  activated?
Can you make a video to see the problem?

**Comment by 0xym** [comment]
I have double checked and "Clear at closing" is activated.
I'm able to reproduce this scenario only when I haven't switched my entry method to Magikeyboard before second database unlock.
After performing steps 1-3 I can:
4. Open DB
5. Switch to Magikeyboard (entry is available)
6. Switch back to regular entry method.
7. Lock DB

And I can repeat steps 4-7 as long as I want providing that I don't navigate to an entry in DB, or open Magikeyboard when DB is locked.

**Comment by J-Jamet** [comment]
Yeah I just reproduced the problem too, it only happens when the Magikeyboard is not the default keyboard when all the procedure is done.
This must happen because the Magikeyboard cannot capture the broadcast when it is not the default keyboard but I don't know how to capture it when it is in the background.

**Comment by J-Jamet** [comment]
I may have a solution: 
When the keyboard is required, check if the database is open (already the case) but also check if the database opening timestamp corresponds to a date earlier than the populate timestamp of magikeyboard entry.
If not, delete the entry in the keyboard.
